pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ClientConfigurationData.java

```diff
@@ -295,6 +295,19 @@
     @JsonIgnore
     private Clock clock = Clock.systemDefaultZone();
 
+    @ApiModelProperty(
+            name = "dnsLookupBindAddress",
+            value = "The Pulsar client dns lookup bind address."
+    )
+    private String dnsLookupBindAddress = null;
+
+    @ApiModelProperty(
+            name = "dnsLookupBindPort",
+            value = "The Pulsar client dns lookup bind port, takes effect when dnsLookupBindAddress is configured." +
+                    " default value is 0."
+    )
+    private int dnsLookupBindPort = 0;
```

Comment:
What does it mean to default to 0 here? From what I can tell, the `InetSocketAddress` will throw an `IllegalArgumentException` when the port is 0.

```java
    private static int checkPort(int port) {
        if (port < 0 || port > 0xFFFF)
            throw new IllegalArgumentException("port out of range:" + port);
        return port;
    }
```

Reply:
@michaeljmarshall 
it will throw exception when port < 0

Reply:
0 means a random port

Reply:
Sorry, I completely misread that if statement! Thanks for clarifying. :)